agent/task.go

```diff
@@ -203,6 +210,12 @@ func (tm *taskManager) run(ctx context.Context) {
 			if cancel != nil {
 				// cancel outstanding operation.
 				cancel()
```

Comment:
This PR looks good and it will likely make the task operations more robust. So +1.

I am glad to see you also raised an issue to docker daemon. I agree that it would be better if docker could provide a way to cancel ongoing container operations like create.